fhevm/tee_bit.go

```diff
@@ -0,0 +1,145 @@
+package fhevm
+
+import (
+	"github.com/ethereum/go-ethereum/common"
+	"go.opentelemetry.io/otel/trace"
+)
+
+func teeShlRun(environment EVMEnvironment, caller common.Address, addr common.Address, input []byte, readOnly bool, runSpan trace.Span) ([]byte, error) {
+	return doShiftOperationGeneric(environment, caller, input, runSpan, func(a, b, typ uint64) uint64 {
+		switch typ {
```

Comment:
For better code readability, type shouldn't be uint64, but tfhe.TfheUintType. Then we switch over Bool, Uint8 etc